### ISSUE TITLE
Style/CommentedKeyword should be aware of RDoc Directives

### ISSUE THREAD

**Issue by halostatue** [rubocop-hq/rubocop#6708]
The following “comment” is an RDoc directive. It is *required* to be on the same line as the `def`.

```
class Foo
  def initialize # :yields self:
    yield self if block_given?
  end
end
```

--------

## Expected behavior

RuboCop should not warn about commented keywords.

## Actual behavior

RuboCop warns about commented keywords.

## Steps to reproduce the problem

Run Rubocop over the following code:

```
class Foo
  def initialize # :yields self:
    yield self if block_given?
  end
end
```

## RuboCop version

`0.63.1 (using Parser 2.5.1.0, running on ruby 2.6.0 x86_64-darwin18)`

**Comment by Drenmi** [comment]
Generally we try not to hard code considerations for other libraries into RuboCop, but we do (arbitrarily) make exceptions for popular libraries. Looking at the cop, we're actually already checking for `:nodoc:`, so it seems reasonable that we'd check for all [RDoc directives](https://docs.ruby-lang.org/en/2.1.0/RDoc/Markup.html#class-RDoc::Markup-label-Directives).

**Comment by bquorning** [comment]
> it seems reasonable that we'd check for all [RDoc directives](https://docs.ruby-lang.org/en/2.1.0/RDoc/Markup.html#class-RDoc::Markup-label-Directives).

In the documentation, the only examples I see of inline comments are for `:nodoc:` and `:yields:`. @halostatue Are you aware of whether any other RDoc directive can be used in an inline/trailing comment?

**Comment by halostatue** [comment]
Not that I can find in my code. It may be worth noting that `# :yields [args]:` is only valid after `def`.

**Comment by bquorning** [comment]
Proposed fix here: #6735

**Comment by koic** [comment]
@halostatue I'm looking at the RDoc official document, the example of `yields` directive looks like `# yields: index, position` (not `# :yields index, position:`). 
https://docs.ruby-lang.org/en/2.1.0/RDoc/Markup.html#class-RDoc::Markup-label-Directives

I will merge the proposal #6735 that fix in compliance with the RDoc official document.

BTW, Thanks always for your artifact diff-lcs gem :-)
